source/h1_connection.c

```diff
@@ -677,45 +696,55 @@ static void s_stream_write_outgoing_data(struct h1_stream *stream, struct aws_io
                         AWS_LS_HTTP_STREAM,
                         "id=%p: Cannot fit any more body data in this message.",
                         (void *)&stream->base);
-                    return;
+                    /* Return success because we want to try again later */
+                    return AWS_OP_SUCCESS;
                 }
 
-                size_t prev_len = dst->len;
-
-                enum aws_http_outgoing_body_state state =
-                    stream->base.stream_outgoing_body(&stream->base, dst, stream->base.user_data);
+                struct aws_stream_status status;
+                aws_input_stream_get_status(stream->base.outgoing_body, &status);
+                if (!status.is_valid) {
+                    AWS_LOGF_ERROR(
+                        AWS_LS_HTTP_STREAM, "id=%p: Body stream reporting status invalid", (void *)&stream->base);
+                    return AWS_OP_ERR;
```

Comment:
Need to raise an error if aws_input_stream_get_status() succeeds, but `status.is_valid` is false

Reply:
Will raise `AWS_IO_STREAM_READ_FAILED`